Lib/test/test_zlib_jy.py

```diff
@@ -28,6 +28,44 @@ def _test_array(self, compress, decompress):
         compressed = array('c', compress('jython'))
         self.assertEqual('jython', decompress(compressed))
 
+    def test_decompress_gzip(self):
+        co = zlib.compressobj(wbits=31)  # window 15 with gzip wrapper.
+        c = co.compress("Jenny: 867-5309")
+        c += co.flush()
+        dco = zlib.decompressobj(wbits=31)
+        d = dco.decompress(c)
+        self.assertEqual(b'',dco.unused_data, msg="dco.unused_data not empty after decompress.")
+        self.assertEqual(b'',dco.unconsumed_tail, msg="dco.unconsumed_tail not empty after decompress.")
+        self.assertEqual("Jenny: 867-5309",d)
+
+    def test_decompress_badlen(self):
+        """Manipulating last two bytes to create invalid initial size check.
```

Comment:
Documentation strings on test functions pop out on the console when that test runs, and long ones come out as a real mess. I converted these to comments.